File: domain/src/main/java/com/ahn/domain/calculator/usecase/GetCalculatorHistoriesUseCase.kt
package com.ahn.domain.calculator.usecase

import com.ahn.domain.calculator.model.CalculatorHistory
import com.ahn.domain.calculator.repository.CalculatorHistoryRepository
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class GetCalculatorHistoriesUseCase @Inject constructor(private val repository: CalculatorHistoryRepository) {
    operator fun invoke(): Flow<List<CalculatorHistory>> {
        return repository.getHistories()
    }
}

File: ios/Packages/Domain/Sources/Domain/Calculator/UseCase/GetCalculatorHistoriesUseCase.swift
public struct GetCalculatorHistoriesUseCase: Sendable {
    private let repository: any CalculatorHistoryRepository

    public init(repository: any CalculatorHistoryRepository) {
        self.repository = repository
    }

    public func callAsFunction() -> AsyncStream<[CalculatorHistory]> {
        repository.histories()
    }
}
